frontend/database.go

```diff
@@ -2,38 +2,127 @@ package main
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"os"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
 )
 
-var (
-	databaseURL  string = os.Getenv("DB_URL")
-	databaseName string = os.Getenv("DB_NAME")
+const (
+	clustersContainer = "Clusters"
 )
 
+// DBClient defines the needed values to perform CRUD operations against the async DB
+type DBClient struct {
+	client *azcosmos.Client
+	DBName string
+	DBUrl  string
+}
+
+// NewDatabaseClient instanstiates a Cosmos DatabaseClient targeting Frontends async DB
+func NewDatabaseClient() *DBClient {
+	cred, _ := azidentity.NewDefaultAzureCredential(nil)
+	d := &DBClient{
+		DBName: os.Getenv("DB_NAME"),
+		DBUrl:  os.Getenv("DB_URL"),
+	}
+	client, _ := azcosmos.NewClient(d.DBUrl, cred, nil)
+	d.client = client
+
+	return d
+}
+
 // DBConnectionTest checks the async database is accessible on startup
-func DBConnectionTest() (string, error) {
-	if databaseName == "none" || databaseName == "" {
+func (d *DBClient) DBConnectionTest() (string, error) {
+	if d.DBName == "none" || d.DBName == "" {
 		return "No database configured, skipping", nil
 	}
 
-	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	database, err := d.client.NewDatabase(d.DBName)
 	if err != nil {
 		return "", err
 	}
-	client, err := azcosmos.NewClient(databaseURL, cred, nil)
+	result, err := database.Read(context.TODO(), nil)
 	if err != nil {
 		return "", err
 	}
-	database, err := client.NewDatabase(databaseName)
+	return result.DatabaseProperties.ID, nil
+}
+
+// GetCluster retreives a cluster document from async DB using resource ID
+func (d *DBClient) GetClusterDoc(partitionKey string) (*HCPOpenShiftClusterDocument, bool, error) {
```

Comment:
There is a context created at the start of each ARM handle function, for example:

```
func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request *http.Request) {
	var err error

	ctx := request.Context()
```

Would be pretty easy to just pass that to the call and re-use it. No objections there unless there is some concern with doing that I'm not aware of

Reply:
Yes, passing the request's context would be appropriate.  To be honest, I'm not sure why it's important other than it seems to be good form in Go-land for functions that need a context to take one rather than create one.

Reply:
Agreed, completely missed the context definition earlier haha. Thanks!